skyagi-web/src/lib/AgentForm.svelte

```diff
@@ -1,18 +1,74 @@
 <script lang="ts">
+	import { error } from '@sveltejs/kit';
+	import { isAgentFormEditing } from './stores';
 	import type { AgentDataType } from './types';
 	import { Label, Input, Button } from 'flowbite-svelte';
+	import { goto } from '$app/navigation';
 
 	export let agentData: AgentDataType = {
+		id: '',
 		name: '',
 		age: '',
 		personalities: '',
 		socialStatus: '',
 		memories: ['']
 	};
 
-	export let handleSubmit = () => {
-		console.log('submitted!');
-	};
+	async function handleSubmit() {
+		if ($isAgentFormEditing) {
```

Comment:
You don't need to put everything here. What if we need different UI logic for create and edit after clicking Submit? This part will grow large. I can refactor later.

Reply:
So pass in different `handleSubmit` function into `AgentForm` component from parent pages, right?

Reply:
Feel free to refactor this if you have better idea.